src/customFileParser/main.ts

```diff
@@ -3,77 +3,124 @@ import Sector from './../Sector';
+    colorMap: [],
+    nodeIdTreeIndexMap: new Map(),
+    idToSectorMap: {},
+  };
+  const compressedData: PerSectorCompressedDataDictionary = {};
+  let uncompressedValues: undefined | UncompressedValues;
+  let rootSector: undefined | Sector;
+
+  sectorBuffers.forEach(sectorBuffer => {
+    const fileReader = new CustomFileReader(sectorBuffer);
+    const sectorByteLength = fileReader.readUint32();
+    const sectorMetadata = fileReader.readSectorMetadata();
     const sector = new Sector(sectorMetadata.sectorBBoxMin, sectorMetadata.sectorBBoxMax);
-    idToSectorMap[sectorMetadata.sectorId] = sector;
+    maps.idToSectorMap[sectorMetadata.sectorId] = sector;
 
-    if (rootSector === undefined || uncompressedValues === undefined) {
+    if (sectorMetadata.arrayCount > 0) { // Is root sector
       rootSector = sector;
       uncompressedValues = fileReader.readUncompressedValues();
-      preloadMeshFiles(meshLoader, uncompressedValues.fileId!);
     } else {
-      const parentSector = idToSectorMap[sectorMetadata.parentSectorId];
+      const parentSector = maps.idToSectorMap[sectorMetadata.parentSectorId];
       if (parentSector !== undefined) {
         parentSector.addChild(sector);
         parentSector.object3d.add(sector.object3d);
-      } else { throw Error('Parent sector not found'); }
-    }
-    const compressedGeometryData = fileReader.readCompressedGeometryData(sectorStartLocation + sectorByteLength);
-    sectorPathToPrimitiveData[sector.path] = compressedGeometryData.primitives;
-    if (compressedGeometryData.instancedMesh) {
-      sectorPathToInstancedMeshData[sector.path] = compressedGeometryData.instancedMesh;
-    }
-    if (compressedGeometryData.mergedMesh) {
-      sectorPathToMergedMeshData[sector.path] = compressedGeometryData.mergedMesh;
+      } else { throw Error('Did not find parent sector'); }
     }
+
+    compressedData[sector.path] = fileReader.readCompressedGeometryData(sectorByteLength);
+  });
+
+  if (rootSector === undefined || uncompressedValues === undefined) {
+    throw Error('Did not find root sector');
   }
 
-  unpackPrimitives(rootSector!, uncompressedValues!, sectorPathToPrimitiveData,
-    treeIndexNodeIdMap, colorMap, filterOptions);
-  unpackMergedMeshes(rootSector!, uncompressedValues!, sectorPathToMergedMeshData, sceneStats,
-    treeIndexNodeIdMap, colorMap);
-  unpackInstancedMeshes(rootSector!, uncompressedValues!, sectorPathToInstancedMeshData, sceneStats,
-    treeIndexNodeIdMap, colorMap);
+  return unpackData(rootSector, uncompressedValues, compressedData, maps, filterOptions);
+}
+
+function unpackData(
+  rootSector: Sector,
+  uncompressedValues: UncompressedValues,
+  compressedData: PerSectorCompressedDataDictionary,
+  maps: DataMaps,
+  filterOptions?: FilterOptions,
+  ): ParseReturn {
+  const sceneStats: SceneStats = {
+    numInstancedMeshes: 0,
+    numMergedMeshes: 0,
+  };
+  unpackPrimitives(rootSector!, uncompressedValues!, compressedData, maps, filterOptions);
```

Comment:
None of the non-null assertions in this function are needed,

Reply:
Oops, forgot to take those out. Thanks